inc/Engine/Preload/Fonts.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace WP_Rocket\Engine\Preload;
+
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+/**
+ * Fonts preload.
+ *
+ * @since 3.6
+ */
+class Fonts implements Subscriber_Interface {
+
+	/**
+	 * WP Rocket Options instance.
+	 *
+	 * @since 3.6
+	 *
+	 * @var Options_Data
+	 */
+	private $options;
+
+	/**
+	 * Return an array of events that this subscriber wants to listen to.
+	 *
+	 * @since  3.6
+	 *
+	 * @param Options_Data $options WP Rocket Options instance.
+	 */
+	public function __construct( Options_Data $options ) {
+		$this->options = $options;
+	}
+
+	/**
+	 * Return an array of events that this subscriber wants to listen to.
+	 *
+	 * @since  3.6
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'wp_head' => [ 'preload_fonts' ],
+		];
+	}
+
+	/**
+	 * Add the required <link/> tags used to preload fonts.
+	 *
+	 * @since 3.6
+	 */
+	public function preload_fonts() {
```

Comment:
2 things here:
- The preload links should be inserted only if the page is cached
- The links should use the CDN URL(s) if CDN is enabled

Reply:
> The preload links should be inserted only if the page is cached
Before to lake this change, please wait our confirmation, we are discussing about that => https://github.com/wp-media/wp-rocket/issues/2543#issuecomment-617788770

Reply:
>The links should use the CDN URL(s) if CDN is enabled

Done in https://github.com/wp-media/wp-rocket/pull/2563/commits/1c5bd1acc576a024e57dc1731579d290a77011be.

Reply:
Please keep the actual behavior to also apply the loading time improvement on uncached version.

Reply:
@Tabrisrp So I guess there is nothing to change here, we don’t need to use `rocket_buffer`.